src/T2DMap.cpp

```diff
@@ -687,37 +685,32 @@ inline void T2DMap::drawRoom(QPainter& painter,
         }
     }
 
-    const bool isRoomSelected = (mPick && roomClickTestRectangle.contains(mPHighlight)) || mMultiSelectionSet.contains(currentRoomId);
-    QLinearGradient selectionBg(roomRectangle.topLeft(), roomRectangle.bottomRight());
-    selectionBg.setColorAt(0.25, roomColor);
-    selectionBg.setColorAt(1, Qt::blue);
+    const bool roomSelected = (mPick && roomClickTestRectangle.contains(mPHighlight)) || mMultiSelectionSet.contains(currentRoomId);
 
-    QPen roomPen(Qt::transparent);
-    roomPen.setWidth(borderWidth);
     painter.setBrush(roomColor);
 
     if (shouldDrawBorder && mRoomWidth >= 12) {
-        roomPen.setColor(mpHost->mRoomBorderColor);
+        pen.setColor(mpHost->mRoomBorderColor);
     } else if (shouldDrawBorder) {
         auto fadingColor = QColor(mpHost->mRoomBorderColor);
         fadingColor.setAlpha(255 * (mRoomWidth / 12));
-        roomPen.setColor(fadingColor);
-    }
-
-    if (isRoomSelected) {
+        pen.setColor(fadingColor);
+    } else if (roomSelected) {
         QLinearGradient selectionBg(roomRectangle.topLeft(), roomRectangle.bottomRight());
         selectionBg.setColorAt(0.2, roomColor);
         selectionBg.setColorAt(1, Qt::blue);
-        roomPen.setColor(QColor(255, 50, 50));
+        pen.setColor(QColor(255, 50, 50));
         painter.setBrush(selectionBg);
+    } else {
+        pen.setColor(Qt::transparent);
     }
```

Comment:
I'm not sure the logic has stayed the same here - it looks as though you are combining the boarder drawing with the room selection drawing - and I'm not entirely sure that is the case. OTOH I didn't do the room borders so I might be missing something...

:confused: